src/main/java/org/wso2/securevault/secret/SecretManager.java

```diff
@@ -282,6 +354,197 @@ private boolean validatePasswords(String identityStorePass,
     }
 
     public String getGlobalSecretProvider() {
+
         return globalSecretProvider;
     }
+
+    /**
+     * Get all the provider listed under both secretRepositories and secretProviders properties.
+     *
+     * @param secretConfigurationProperties All the configuration properties.
+     */
+    private void getAllProviders(Properties secretConfigurationProperties) {
+
+        readLegacyProviders(secretConfigurationProperties);
+        readNovelProviders(secretConfigurationProperties);
+    }
+
+    /**
+     * Read the providers listed under secretRepositories property.
+     *
+     * @param secretConfigurationProperties All the configuration properties.
+     */
+    private void readLegacyProviders(Properties secretConfigurationProperties) {
+
+        String legacyProvidersString =
+                getPropertiesFromSecretConfigurations(secretConfigurationProperties, PROP_SECRET_REPOSITORIES);
+
+        if (isPropValueValidated(legacyProvidersString)) {
+            isLegacyProvidersExists = true;
+            String[] legacyProviders = addStringToArray(legacyProvidersString);
+            addToProvidersMap(legacyProviders, PROP_SECRET_REPOSITORIES);
+        }
+    }
+
+    /**
+     * Read the providers listed under secretProviders property.
+     *
+     * @param secretConfigurationProperties All the configuration properties.
+     */
+    private void readNovelProviders(Properties secretConfigurationProperties) {
+
+        String novelProvidersString =
+                getPropertiesFromSecretConfigurations(secretConfigurationProperties, PROP_SECRET_PROVIDERS);
+
+        if (isPropValueValidated(novelProvidersString)) {
+            isNovelProvidersExists = true;
```

Comment:
isNovelProvidersExists doesn't sound correct :)

Reply:
Changed to novelProvidersExist